File: windows/src/AgentIsland/Model/CodexJumpPreferenceStore.cs
using System.ComponentModel;
using AgentIsland.Core;

namespace AgentIsland.Model;

/// How "Open thread" treats a Codex session: deep-link into the desktop
/// app's exact thread (default — codex://threads/<id> is the app's own
/// official route), or reopen the conversation in a terminal via
/// `codex resume`, for people who live in the CLI. (Parity with macOS.)
public sealed class CodexJumpPreferenceStore : INotifyPropertyChanged
{
    public static CodexJumpPreferenceStore Shared { get; } = new();

    private const string Key = "AgentIsland.codexJumpPrefersCli";

    private bool _prefersCli;

    public event PropertyChangedEventHandler? PropertyChanged;

    private CodexJumpPreferenceStore()
    {
        _prefersCli = Preferences.Get<bool?>(Key) ?? false;
    }

    public bool PrefersCli
    {
        get => _prefersCli;
        set
        {
            if (_prefersCli == value) return;
            _prefersCli = value;
            Preferences.Set(Key, value);
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(PrefersCli)));
        }
    }
}

File: Sources/Model/CodexJumpPreferenceStore.swift
import Foundation

/// How "Open thread" treats a Codex session: deep-link into the desktop
/// app's exact thread (default — `codex://threads/<id>` is the app's own
/// official route), or reopen the conversation in a terminal via
/// `codex resume`, for people who live in the CLI.
@MainActor
final class CodexJumpPreferenceStore: ObservableObject {
    static let shared = CodexJumpPreferenceStore()

    private static let key = "AgentIsland.codexJumpPrefersCLI"

    @Published var prefersCLI: Bool {
        didSet { UserDefaults.standard.set(prefersCLI, forKey: Self.key) }
    }

    private init() {
        // The picker is gone from Settings (2026-07-14) — desktop app is
        // the one behavior. The stored value is ignored, not erased, so
        // the picker can return without losing anyone's old choice.
        prefersCLI = false
    }
}
